packages/orchestration/src/types.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-use-before-define -- fine for types */
+// Ambients
+import '@agoric/zoe/exported.js';
+
+import type { Amount, Brand } from '@agoric/ertp/exported.js';
+import type {
+  CoinAmount,
+  CosmosChainInfo,
+  CosmosStakingFacet,
+  Denom,
+  IcaAccount,
+} from './cosmos-api.js';
+import type { EthChainInfo } from './ethereum-api.js';
+import type { ICQConnection } from './exos/icqConnectionKit.js';
+import type {
+  ChainAddress,
+  OrchestrationAccountI,
+  TransferMsg,
+} from './orchestration-api.js';
+
+export type * from './cosmos-api.js';
+export type * from './exos/chainAccountKit.js';
+export type * from './exos/icqConnectionKit.js';
+export type * from './orchestration-api.js';
+export type * from './service.js';
+export type * from './vat-orchestration.js';
+
+/**
+ * static declaration of known chain types will allow type support for
+ * additional chain-specific operations like `liquidStake`
+ */
+export type KnownChains = {
+  stride: {
+    info: CosmosChainInfo;
+    methods: {
+      liquidStake: (amount: AmountArg) => Promise<void>;
+      // ??? do we prefer to merge capabilities into a single object? If we do
+      // that, then the chain-native methods (e.g. delegate) are hard to
+      // distinguish from generic. Alternately, there could be a method to get a
+      // generic facade on the chain-native methods.
+      getIcaAcccount: () => IcaAccount;
+      getIcqConnection: () => ICQConnection;
+      getStakingFacet: () => CosmosStakingFacet;
+    };
+  };
+  cosmos: { info: CosmosChainInfo; methods: {} };
+  agoric: {
+    info: Omit<CosmosChainInfo, 'ibcConnectionInfo'>;
+    methods: {
+      // TODO reference type from #8624 `packages/vats/src/localchain.js`
+      /**
+       * Register a hook to intercept an incoming IBC Transfer and handle it.
+       * Calling without arguments will unregister the hook.
+       */
+      interceptTransfer: (tap?: {
+        upcall: (args: any) => Promise<any>;
+      }) => Promise<void>;
+    };
+  };
+  celestia: {
+    info: CosmosChainInfo;
+    methods: {
+      // TODO DRY by feature flag of application modules enabled on chain
+      getIcaAcccount: () => IcaAccount;
+      getStakingFacet: () => CosmosStakingFacet;
+    };
+  };
+  osmosis: { info: CosmosChainInfo; methods: {} };
+};
+
+/**
+ * In many cases, either a denom string or a local Brand can be used to
+ * designate a remote token type.
+ */
+export type DenomArg = Brand | Denom;
+
+/** Amounts can be provided as pure data using denoms or as native Amounts */
+export type AmountArg = CoinAmount | Amount;
```

Comment:
Nit: Order of "agoric type | low level type" should match between DenomArg and AmountArg.

Reply:
For usability, I expect the low-level type should be first.

Reply:
I read this as a the request:

>  "agoric type | low level type"

but you want it the other way? I agree it's better to have low level type first, but I want to confirm your request before I make another edit